eng/pipelines/aggregate-reports.yml

```diff
@@ -88,6 +87,9 @@ jobs:
       condition: succeededOrFailed()
       displayName: 'Publish Report Artifacts'
       artifact: reports
+
+    - template: ../steps/upload-repository-on-failure.yml
```

Comment:
Were we uploading the cache on failure before? I think it would be important to determine when the failure occurs as the Maven cache could be in different states (missing artifacts that were going to be used but were never downloaded)

Reply:
This doesn't actually upload the cache it still does what it did before which is upload the repo only but this template also now does the cache clean-up step as well.